python_modules/libraries/dagster-fivetran/dagster_fivetran_tests/beta/test_asset_specs.py

```diff
@@ -31,6 +44,113 @@ def test_fetch_fivetran_workspace_data(
+        (None, None, {"non_matching_database"}, None, 0, True, True),
+        (None, None, None, {"non_matching_service"}, 0, True, True),
+        (
+            {"non_matching_connector_name"},
+            {"non_matching_connector_id"},
+            {"non_matching_database"},
+            {"non_matching_service"},
+            0,
+            True,
+            True,
+        ),
+        (
+            {TEST_CONNECTOR_NAME},
+            {TEST_CONNECTOR_ID},
+            {"non_matching_database"},
+            None,
+            0,
+            True,
+            True,
+        ),
+        (
+            {"non_matching_connector_name"},
+            None,
+            {TEST_DESTINATION_DATABASE},
+            {TEST_DESTINATION_SERVICE},
+            0,
+            False,
+            True,
+        ),
+    ],
+    ids=[
+        "no_filter_present_connector",
+        "connector_name_filter_present_connector",
+        "connector_id_filter_present_connector",
+        "database_filter_present_connector",
+        "service_filter_present_connector",
+        "all_filters_present_connector",
+        "connector_name_filter_absent_connector",
+        "connector_id_filter_absent_connector",
+        "database_filter_absent_connector",
+        "service_filter_absent_connector",
+        "all_filters_absent_connector",
+        "one_non_matching_database_filter_absent_connector",
+        "one_non_matching_connector_filter_absent_connector",
+    ],
+)
+def test_fivetran_filter(
+    connector_names: Optional[Set[str]],
+    connector_ids: Optional[Set[str]],
+    databases: Optional[Set[str]],
+    services: Optional[Set[str]],
+    expected_result: int,
+    skip_connector_call: bool,
+    skip_schema_config_call: bool,
+    fetch_workspace_data_api_mocks: responses.RequestsMock,
+) -> None:
+    resource = FivetranWorkspace(
+        account_id=TEST_ACCOUNT_ID, api_key=TEST_API_KEY, api_secret=TEST_API_SECRET
+    )
+
+    # When a destination is filtered out, all its connector are filtered out
+    # and calls to the `/groups/{group_id}/connectors` endpoint are skipped.
+    # We remove the call from the mock calls to avoid the exception
+    # raised by RequestMock when not all requests are executed.
+    if skip_connector_call:
+        fetch_workspace_data_api_mocks.remove(
+            method_or_response=responses.GET,
+            url=f"{FIVETRAN_API_BASE}/{FIVETRAN_API_VERSION}/groups/{TEST_DESTINATION_ID}/connectors",
+        )
+    # When a connector is filtered out, the call to the `connectors/{connector_id}/schemas` endpoint is skipped.
+    # We remove the call from the mock calls to avoid the exception
+    # raised by RequestMock when not all requests are executed.
+    if skip_schema_config_call:
+        fetch_workspace_data_api_mocks.remove(
+            method_or_response=responses.GET,
+            url=f"{get_fivetran_connector_api_url(TEST_CONNECTOR_ID)}/schemas",
+        )
```

Comment:
Handling these skipped calls is only required for this test, so removing the mocked calls here based on the parametrization seemed like the most logical approach. Open to other suggestions.